src/Stratis.Bitcoin.Features.Consensus/Rules/ProvenHeaderRules/ProvenHeaderSizeRule.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.ProvenHeaderRules
+{
+    /// <summary>
+    /// Rule to check if the serialized sizes of the proven header components, such as merkle proof (max 512 bytes),
+    /// signature (max 80 bytes) and coinstake (max 1,000,000 bytes), do not exceed maximum possible size allocation.
+    /// </summary>
+    /// <seealso cref="ProvenHeaderValidationConsensusRuleBase" />
+    public class ProvenHeaderSizeRule : ProvenHeaderValidationConsensusRuleBase
```

Comment:
`ProvenHeaderValidationConsensusRuleBase` wait what?

Reply:
We probably can make it shorter: `ProvenHeaderRuleBase`. Sounds good?